### PR TITLE
Keras 3 movielens examples

### PR DESCRIPTION
**Pull request by mattdangerw**
tf-only cause of some funky and inscrutable preprocessing. data parsing wasn't working, but through some pandas flags at it till it resolved.

### CODE REVIEW

File: examples/structured_data/md/movielens_recommendations_transformers.md [hunk 1]
```diff
@@ -577,16 +572,21 @@ print(f"Test MAE: {round(rmse, 3)}")
 <div class="k-default-codeblock">
 ```
 Epoch 1/5
-1598/1598 [==============================] - 46s 27ms/step - loss: 1.6617 - mean_absolute_error: 0.9981
+ 1600/1600 ━━━━━━━━━━━━━━━━━━━━ 19s 11ms/step - loss: 1.5762 - mean_absolute_error: 0.9892
 Epoch 2/5
-1598/1598 [==============================] - 43s 27ms/step - loss: 1.0282 - mean_absolute_error: 0.8101
+   16/1600 [37m━━━━━━━━━━━━━━━━━━━━  10s 7ms/step - loss: 1.1720 - mean_absolute_error: 0.8726
+
+/home/matt/miniconda3/envs/keras-nlp-tensorflow/lib/python3.10/contextlib.py:153: UserWarning: Your input ran out of data; interrupting training. Make sure that your dataset or generator can generate at least `steps_per_epoch * epochs` batches. You may need to use the `.repeat()` function when building your dataset.
```
Comment (fchollet) [inline examples/structured_data/md/movielens_recommendations_transformers.md:579]: Please scrub this line (manually)

Comment (mattdangerw) [reply examples/structured_data/md/movielens_recommendations_transformers.md]: done

**Review by fchollet — APPROVED** [review]
LGTM, thank you
